packages/puppeteer-core/src/common/util.ts

```diff
@@ -365,6 +365,13 @@ export function addPageBinding(type: string, name: string): void {
   // @ts-expect-error: In a different context.
   const callCdp = globalThis[name];
 
+  // Depending on the frame loading state either Runtime.evaluate or
+  // Page.addScriptToEvaluateOnNewDocument might succeed. Let's check that we
+  // don't re-wrap Puppeteer's binding.
+  if (callCdp.toString().includes('This is the Puppeteer binding.')) {
```

Comment:
I don't like looking for a specific Comment in the function. How about we do this after:
```ts
Object.assign(globalThis[name],{
  toString(): {
    return 'PuppeteerBinding'
  }
}
```
And this turns into
```suggestion
  if (callCdp.toString() === 'PuppeteerBinding') {
```

Reply:
Not sure, it might be confusing for debugging if the function toString() returns something else. But we can tag it in a different way.

Reply:
Yeah that's true, lets then modify the `toStringTag` as its
```ts
Object.assign(globalThis[name],{
  [Symbol.toStringTag]: 'PuppeteerBinding',
}
```
```ts
if (callCdp[Symbol.toStringTag] === 'PuppeteerBinding') {
```
![image](https://github.com/puppeteer/puppeteer/assets/34244704/bd01afe6-010a-4e0c-9778-84e01eef9e35)